File: backend-ms1/src/main/java/ma/zs/zyn/service/facade/admin/vocab/VocabularyQuizAdminService.java
package ma.zs.zyn.service.facade.admin.vocab;

import java.util.List;
import ma.zs.zyn.bean.core.vocab.VocabularyQuiz;
import ma.zs.zyn.dao.criteria.core.vocab.VocabularyQuizCriteria;
import ma.zs.zyn.zynerator.service.IService;



public interface VocabularyQuizAdminService extends  IService<VocabularyQuiz,VocabularyQuizCriteria>  {

    List<VocabularyQuiz> findBySectionId(Long id);
    int deleteBySectionId(Long id);
    long countBySectionCode(String code);




}

File: frontend/src/app/shared/service/admin/vocab/VocabularyQuizAdmin.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {BehaviorSubject, Observable} from 'rxjs';

import {environment} from 'src/environments/environment';

import {VocabularyQuizDto} from 'src/app/shared/model/vocab/VocabularyQuiz.model';
import {VocabularyQuizCriteria} from 'src/app/shared/criteria/vocab/VocabularyQuizCriteria.model';
import {AbstractService} from 'src/app/zynerator/service/AbstractService';


@Injectable({
  providedIn: 'root'
})
export class VocabularyQuizAdminService extends AbstractService<VocabularyQuizDto, VocabularyQuizCriteria> {
     constructor(private http: HttpClient) {
        super();
        this.setHttp(http);
    }

    public constrcutDto(): VocabularyQuizDto {
        return new VocabularyQuizDto();
    }

    public constrcutCriteria(): VocabularyQuizCriteria {
        return new VocabularyQuizCriteria();
    }

    get API() {
        return environment.apiUrlZynservice + 'admin/vocabularyQuiz/';
    }
}
